components/tag-list/tag-list.js

```diff
@@ -133,9 +134,13 @@ class TagList extends LocalizeCoreElement(InteractiveMixin(ArrowKeysMixin(LitEle
 		});
 		this._clearButtonResizeObserver.observe(clearButton);
 
-		const container = this.shadowRoot.querySelector('.tag-list-outer-container');
-		this._resizeObserver = new ResizeObserver((e) => requestAnimationFrame(() => this._handleResize(e)));
-		this._resizeObserver.observe(container);
+		const offsetParent = getOffsetParent(this);
+		this._resizeObserver = new ResizeObserver((e) => requestAnimationFrame(() => {
+			if (offsetParent.tagName === 'BODY') this._availableWidth = Math.floor(this.parentNode.offsetWidth);
```

Comment:
I put this `if/else` block here because it's easy to access `offsetParent` here for this check, but it could make sense to do a bit of refactoring and put all the logic back in `_handleResize`